unlock-app/src/components/interface/checkout/main/Confirm/ConfirmUniversalCard.tsx

```diff
@@ -0,0 +1,231 @@
+import { CheckoutService } from './../checkoutMachine'
+import { Connected } from '../../Connected'
+import { useConfig } from '~/utils/withConfig'
+import { Button } from '@unlock-protocol/ui'
+import { Fragment, useState } from 'react'
+import { loadStripeOnramp } from '@stripe/crypto'
+import { useActor } from '@xstate/react'
+import { storage } from '~/config/storage'
+
+import { PoweredByUnlock } from '../../PoweredByUnlock'
+import { ViewContract } from '../../ViewContract'
+import { PricingData } from './PricingData'
+import { CryptoElements, OnrampElement } from '../../utils/CryptoElements'
+import { CreditCardPricingBreakdown } from './ConfirmCard'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useUniversalCardPrice } from '~/hooks/useUniversalCardPrice'
+import { usePurchaseData } from '~/hooks/usePurchaseData'
+
+interface Props {
+  injectedProvider: unknown
+  checkoutService: CheckoutService
+  onConfirmed: (lock: string, hash?: string) => void
+  onError: (message: string) => void
+}
+
+export function ConfirmUniversalCard({
+  injectedProvider,
+  checkoutService,
+  onConfirmed,
+  onError,
+}: Props) {
+  const { getWalletService, account } = useAuth()
+  const [state] = useActor(checkoutService)
+  const config = useConfig()
+  const [sessionError, setSessionError] = useState<string>('')
+  const [onrampSession, setOnrampSession] = useState<any>(null)
+  const stripeOnrampPromise = loadStripeOnramp(config.stripeApiKey)
+
+  const { lock, recipients, captcha, paywallConfig, password, promo } =
+    state.context
+
+  // Build the `purchaseData` field that gets passed to the contract
+  const { isInitialLoading: isInitialDataLoading, data: purchaseData } =
+    usePurchaseData({
+      lockAddress: lock!.address,
+      network: lock!.network,
+      promo,
+      password,
+      captcha,
+      paywallConfig,
+      recipients,
+    })
+
+  // And now get the price to pay by card
+  const { data: cardPricing, isInitialLoading: isCardPricingLoading } =
+    useUniversalCardPrice({
+      network: lock!.network,
+      lockAddress: lock!.address,
+      recipients,
+      purchaseData: purchaseData!,
+      enabled: !isInitialDataLoading,
+    })
+
+  // TODO: Also configure webhook on the Stripe side?
+  const onChange = async (event: any) => {
+    const { session } = event
+
+    if (session.wallet_address && session.wallet_address !== account) {
+      setSessionError(
+        'You cannot change the recipient address as it is the address that will receive the NFT membership token. Please start again.'
+      )
+    }
+
+    const expectedAmount = cardPricing!.total
```

Comment:
```suggestion
    const expectedAmount = cardPricing?.total
```
worried about when `cardPricing .total` not actually present here